### PR TITLE
Windows compatibility changes

### PR DESCRIPTION
**Pull request by PertsevRoman**
Windows OS fixes

- installation paths
- tests

### CODE REVIEW

**Comment by PertsevRoman** [comment]
I also want to fix some tests before the PR is being merged

**Comment by stashedUp** [comment]
@Shuliyey  what are your thoughts on @PertsevRoman suggestion?
He is modifying the path for Window's installation. You had mentioned this issue on your "TODO".

**Comment by warrensbox** [comment]
@Shuliyey what are your thoughts on @PertsevRoman suggestion?
He is modifying the path for Window's installation. You had mentioned this issue on your "TODO".

**Comment by Shuliyey** [comment]
morning warren, i see the change is on refactoring code to avoid code duplication through introducing below 3 functions
```go
// get install path variable value  (windows os runtime support)
func getInstallPath() string {
	return string(os.PathSeparator) + installPath + string(os.PathSeparator)
}

// get versioned install filename (windows os runtime support)
func getVersionedInstallFileName(tfversion string) string {
	if runtime.GOOS == "windows" {
		return getInstallLocation() + installVersion + tfversion + ".exe"
	}

	return getInstallLocation() + installVersion + tfversion
}

// get install filename (windows os runtime support)
func getInstallFileName() string {
	if runtime.GOOS == "windows" {
		return getInstallLocation() + installFile + ".exe"
	}

	return installLocation + installFile
}
```

i like the idea, but would conflict to be resolved first (@PertsevRoman) to get better analysis

File: lib/install.go [hunk 1]
```diff
@@ -49,6 +49,29 @@ func initialize() {
...
+		return getInstallLocation() + installVersion + tfversion + ".exe"
+	}
+
+	return getInstallLocation() + installVersion + tfversion
+}
+
+// get install filename (windows os runtime support)
+func getInstallFileName() string {
+	if runtime.GOOS == "windows" {
+		return getInstallLocation() + installFile + ".exe"
```
Comment (Shuliyey) [inline lib/install.go:69]:
```suggestion
		return filepath.Join(getInstallLocation(), installFile + ".exe")
```

File: lib/install.go [hunk 2]
```diff
@@ -49,6 +49,29 @@ func initialize() {
...
 
+// get install path variable value  (windows os runtime support)
+func getInstallPath() string {
+	return string(os.PathSeparator) + installPath + string(os.PathSeparator)
+}
+
+// get versioned install filename (windows os runtime support)
+func getVersionedInstallFileName(tfversion string) string {
+	if runtime.GOOS == "windows" {
+		return getInstallLocation() + installVersion + tfversion + ".exe"
```
Comment (Shuliyey) [inline lib/install.go:60]:
```suggestion
		return filepath.Join(getInstallLocation(), installVersion+ tfversion + ".exe")
```

File: main.go [hunk 3]
```diff
@@ -278,7 +278,7 @@ func retrieveFileContents(file string) string {
 		fmt.Printf("Error: %s\n", err)
 		os.Exit(1)
 	}
-	tfversion := strings.TrimSuffix(string(fileContents), "\n")
```
Comment (warrensbox) [inline main.go:281]: @PertsevRoman why are we trimming the space instead of the new line character?

File: lib/install.go (hunk 1 above)
Comment (warrensbox) [reply lib/install.go]: @Shuliyey looks good.
PertsevRoman  are you able to amend this?

File: lib/install.go (hunk 2 above)
Comment (warrensbox) [reply lib/install.go]: PertsevRoman  are you able to amend this?

**Pull request merged**